### PR TITLE
Remove the legacy osc_compat mode

### PR DESCRIPTION
**Pull request by tas50**
This provides compatibility with Chef Server 11. It's time to remove
this.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
Fixes https://github.com/chef/chef/issues/8116

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
looking over test failures, they're unrelated... but there's a lot, someone should double-check before merging they're not related.

File: lib/chef/chef_fs/config.rb [hunk 1]
```diff
@@ -146,8 +146,8 @@ def initialize(chef_config = Chef::Config, cwd = Dir.pwd, options = {}, ui = nil
 
         if @chef_config[:repo_mode] == "everything" && is_hosted? && !ui.nil?
           ui.warn %Q{You have repo_mode set to 'everything', but your chef_server_url
-              looks like it might be a hosted setup.  If this is the case please use
-              hosted_everything or allow repo_mode to default}
+              looks like it might be Chef Infra Server 12+ or Hosted Chef. If this is
+              the case please use 'hosted_everything' or allow repo_mode to default}
```
Comment (kagarmoe) [inline lib/chef/chef_fs/config.rb:150]:
```suggestion
              the case, use 'hosted_everything' or allow 'repo_mode' to default}
```

**Review by kagarmoe — CHANGES_REQUESTED** [review]
tenor-93260980

**Comment by tas50** [comment]
We have the deprecation which is enough for now

**Pull request closed without merging**